### DISCUSSION TITLE
Build times for individual modules

### DISCUSSION THREAD

**Discussion by leovido** [pointfreeco/swift-composable-architecture#955] [Q&A] [answered]
Hi,

I wanted to check if there's a way of getting a report of how long each module takes to build with TCA.
Currently, I tried adding **Other Swift Flags** to get warnings if a function or expression is taking longer to compile but get nothing on Xcode. I have done this on the `.xcodeproj` itself on **Build Settings**.

```
-Xfrontend -warn-long-function-bodies=100
-Xfrontend -warn-long-expression-type-checking=100
```

The current outcome is that I don't receive any of these warnings. Likely to be because the dependencies have their own space and they are not included on the .xcodeproj file itself.

The expected outcome is either a file with logs with compile times for each file, method, class, etc.
OR
Display the warning messages on Xcode

https://github.com/fastred/Optimizing-Swift-Build-Times

**Comment by stephencelis** [comment, marked as answer]
Hi @kuriishu27! Are you asking how long TCA takes to build? On my machine, a completely clean build (which includes all TCA's dependencies) takes about 6 seconds, and subsequent builds are pretty much instantaneous.

If I add the following build settings to the `ComposableArchitecture` module:

```swift
swiftSettings: [
  .unsafeFlags([
    "-Xfrontend", "-warn-long-expression-type-checking=100",
    "-Xfrontend", "-warn-long-function-bodies=100",
  ])
]
```

I don't get any warnings, because there are no expressions or function bodies that take over 100ms to build.

**Comment by AlexisQapa** [comment]
To complement Stephen answer, If you use SPM to define your targets you can do this
```swift
targets: [
    .target(name: "MyTarget", dependencies: []),
]
.map { (target: Target) in
    target.swiftSettings = [.unsafeFlags([
        "-Xfrontend",
        "-warn-long-function-bodies=100",
        "-Xfrontend",
        "-warn-long-expression-type-checking=100"
    ])]
return target
}
```
which is convenient to set those flags on all or at least multiple targets.

**Comment by leovido** [comment]
I found this extremely helpful and it's displaying the expected results. Thank you!
